source/elements/oneDPL/source/parallel_api/buffer_wrappers.rst

```diff
@@ -43,33 +43,50 @@ Buffer Wrappers
 
 ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` are helper functions
 for passing `SYCL`_ buffers to oneDPL algorithms.
-These functions accept a buffer and return an object
-of an unspecified type that satisfies the following requirements:
-
-- it is ``CopyConstructible``, ``CopyAssignable``, and comparable
-  with operators ``==`` and ``!=``;
-- the following expressions are valid: ``a + n``, ``a - n``,
-  ``a - b``, where ``a`` and ``b`` are objects of the type,
-  and ``n`` is an integer value;
-- it provides the ``get_buffer()`` method that returns the buffer
-  passed to the ``begin`` or ``end`` function.
+These functions accept a SYCL buffer and return an object of an unspecified type
+(referred below as *buffer position*) that satisfies the following requirements:
+
+- It is ``CopyConstructible`` and ``CopyAssignable``.
+- It is comparable with operators ``==`` and ``!=``.
+- It provides the ``get_buffer()`` method that returns the SYCL buffer, over which an object was built.
+- The expressions ``a + n`` and ``a - n``, where ``a`` is a buffer position object and ``n``
+  is an integer value, are valid and evaluate to a buffer position object representing
+  a position in the buffer that follows or precedes the position of ``a`` by ``n``.
```

Comment:
Too many "position" words, though I am not sure how to make it better. Maybe:
```suggestion
  is an integer value, are valid and evaluate to a buffer position object representing
  an offset in the buffer that follows or precedes the position of ``a`` by ``n``.
```

Reply:
I also do not quite like it, and also have no good idea how to improve.

Reply:
location?  index?
also "in the buffer that follows" could be may be better as "into the buffer that follows" if we use "offset" or "index"

Reply:
The reading problem here is really caused by "a buffer position object representing a position in the buffer", which sounds like tautology. But without it, the wording would become sloppy:

> evaluate to a buffer position object that follows or precedes ``a`` by ``n``.

because what is that "follows/precedes" relation between the *objects* (and not positions)? We would need to introduce it first.

 And if we change to "a buffer position object representing an index into the buffer", then why is not it a buffer *index* object? :) Same for "offset", and also offsets do not follow or precede one another. And I would not speak of an element either, because the object gives no direct way to access an element or even to get its index.

Reply:
I think something like "a buffer position object representing an index location into the buffer" would be reasonable and get the point the point across without seeming to define a word with itself.

Reply:
Let me try this:
> ... evaluate to a buffer position object such that the corresponding position in the buffer follows or precedes that of ``a`` by ``n``.